libnmstate/nm/sriov.py

```diff
@@ -43,11 +43,79 @@ def create_setting(iface_state, base_con_profile):
         if not sriov_setting:
             sriov_setting = nmclient.NM.SettingSriov.new()
 
+        vfs_config = sriov_config.get(Ethernet.SRIOV.VFS_SUBTREE, [])
+        vf_object_ids = {vf.get_index() for vf in sriov_setting.props.vfs}
+        vf_config_ids = {
+            vf_config[Ethernet.SRIOV.VFS.ID] for vf_config in vfs_config
+        }
+        _create_sriov_vfs_in_setting(
+            vfs_config, sriov_setting, vf_config_ids - vf_object_ids
+        )
+        _remove_sriov_vfs_in_setting(
+            vfs_config, sriov_setting, vf_object_ids - vf_config_ids
+        )
+
         sriov_setting.props.total_vfs = sriov_config[Ethernet.SRIOV.TOTAL_VFS]
 
     return sriov_setting
 
 
+def _create_sriov_vfs_in_setting(vfs_config, sriov_setting, vf_ids_to_add):
+    for vf_config in vfs_config:
+        vf_already_exists = True
+        vf_id = vf_config[Ethernet.SRIOV.VFS.ID]
+        if vf_id in vf_ids_to_add:
+            vf_already_exists = False
```

Comment:
I am doing this because when editting we need to duplicate the existing VF object, then remove it from the list of VFs and add the one that we have duplicated. If we do not do that, we can generate some unstable situations (random fails).

Reply:
But I am going to split this in different functions to make it easier! Thanks!

Reply:
Done, thanks!